crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1089 @@
+/// uniform color space. This means that the gradient will be perceived as having
+/// a uniform progression of colors. This is particularly useful for data
+/// visualization.
+///
+/// However, you can choose to interpolate the gradient in any supported color
+/// space you want, but beware that some color spaces are not suitable for
+/// perceptually interpolating between colors. Below you can find a list of
+/// color spaces and whether they are suitable for perceptual interpolation.
+///
+/// |           Color space           | Perceptually uniform? |
+/// | ------------------------------- | --------------------- |
+/// |      [Oklab]($color.oklab)      |           ✅          |
+/// |      [sRGB]($color.rgb)         |           ❌          |
+/// | [linear-RGB]($color.linear-rgb) |           ✅          |
+/// |      [CMYK]($color.cmyk)        |           ❌          |
+/// |     [Grayscale]($color.luma)    |           ✅          |
+/// |       [HSL]($color.hsl)         |           ❌          |
+/// |       [HSV]($color.hsv)         |           ❌          |
+///
+/// ```example
+/// #set text(fill: white)
+/// #set block(spacing: 0pt)
+///
+/// #let spaces = (
+///   ("Oklab", color.oklab),
+///   ("sRGB", color.rgb),
+///   ("linear-RGB", color.linear-rgb),
+///   ("CMYK", color.cmyk),
+///   ("Grayscale", color.luma),
+///   ("HSL", color.hsl),
+///   ("HSV", color.hsv),
+/// )
+///
+/// #for space in spaces {
+///   block(
+///     width: 100%,
+///     height: 10pt,
+///     fill: gradient.linear(red, blue, space: space.at(1))
+///   )[
+///     #space.at(0)
+///   ]
+/// }
+/// ```
+///
+/// ## Direction
+///
+/// Some gradients are sensitive to the direction of the gradient. For example, a
+/// linear gradient has an angle that determines the direction of the gradient. Instead
+/// of the traditional clockwise angle, Typst uses an anti-clockwise angle, with 0° being
+/// from left-to-right, 90° from top-to-bottom, 180° from right-to-left, and 270° from
+/// bottom-to-top.
+///
+/// ```example
+/// #set block(spacing: 0pt)
+/// #stack(
+///   dir: ltr,
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 0deg)),
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 90deg)),
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 180deg)),
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 270deg)),
+/// )
+/// ```
+///
+/// ## Note on compatibility
+///
+/// Typst's gradients were designed to be widely compatible; however, in
+/// [PDF.js](https://mozilla.github.io/pdf.js/), the PDF reader bundled with Firefox,
+/// gradients in `rotate` blocks may not be rendered correctly. This is a bug in
+/// PDF.js and not in Typst. Despite this, every type of gradient has been
+/// tested in every major PDF reader, and should work in most browsers as
+/// expected.
+///
+/// ## Presets
+///
+/// Typst also includes a number of preset color maps. In the following section, the
```

Comment:
Is "color maps" an appropriate term for gradients?

Reply:
I designed the gradient PR to be able to specifically do both jobs (for plotting packages) so I would say so, and since those are not gradients, the correct term is color map.